src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,42 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private static final int MAX_SIZE = 10;
+    private int size = 0;
+
+    private Pair[] data = new Pair[MAX_SIZE];
+
     @Override
     public void put(K key, V value) {
+        for (int j = 0; j < size(); j++) {
+            if (data[j] != null && data[j].getKey() != null
+                    && data[j].getKey().equals(key)
+                    || data[j] != null
+                    && (data[j].getKey() == null && key == null)) {
```

Comment:
```suggestion
            if (data[j].getKey() != null && data[j].getKey().equals(key)
                    || data[j].getKey() == key) {
```
в тебе не може бути dara[j] == null якщо ти ітеруєшся тільки до сайзу

Reply:
Исправлю. Спасибо за ревью!)